sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -92,16 +97,24 @@ public boolean accepts(Class<?> extensionType) {
             return (Factory) e.create();
         });
 
+        forEachConfigurerFactory(extensionType, (factory, annotation) ->
+                factory.createForType(annotation, extensionType).accept(config));
+
         Map<Method, Handler> handlers = buildHandlersFor(extensionType, config);
         MethodInterceptor interceptor = createMethodInterceptor(handlers, handle);
         return extensionType.cast(f.newInstance(interceptor));
     }
 
-    private Map<Method, Handler> buildHandlersFor(Class<?> sqlObjectType, SqlObject config) {
+    private Map<Method, Handler> buildHandlersFor(Class<?> sqlObjectType, SqlObjectConfig config) {
         return handlersCache.computeIfAbsent(sqlObjectType, type -> {
 
             final Map<Method, Handler> handlers = new HashMap<>();
             for (Method method : sqlObjectType.getMethods()) {
+                // FIXME will applying configurers here and caching config create problems?
+                SqlObjectConfig methodConfig = config.createCopy();
+                forEachConfigurerFactory(method, (factory, annotation) ->
+                        factory.createForMethod(annotation, sqlObjectType, method).accept(methodConfig));
```

Comment:
We should probably fix this before merge.

In a nutshell, whatever the `SqlObjectConfig` looks like the first time an extension class is created is cached and reused here, effectively making the default config set in stone.

What I'm thinking is that `Handler` should not receive the `SqlObjectConfig` until the invocation. Right now most handlers receive it as a constructor parameter, so this will be a little work to address. But since it's a breaking change I think it's better to do now.